### PR TITLE
Use netplan.io for network management

### PR DESCRIPTION
**Pull request by IsaacJT**
Replace the usage of raw systemd-networkd and udev rules with
netplan.io, which can do the job of both in one YAML file, simplifying
the network configurations.

Additionally, do not remove python3 packages from the rootfs, as these are required
for netplan.io. This particular change actually has very little effect on the disk
usage; python3.10 doesn't exist in lunar (as it uses 3.11), so removing
the package was essentially a no-op before. Removing the python3 package
basically just removes the symlink for python3 to python3.11, which also
doesn't have an impact on the rootfs size.

### CODE REVIEW

**Comment by IsaacJT** [comment]
@schaefi curious what you think about this :)

**Comment by schaefi** [comment]
> @schaefi curious what you think about this :)

So you are using netplan and that's a nice concept as it servers as an abstraction layer for the configuration the actual daemon (in this case systemd-networkd) and the interface rules (udev). That's all nice and I have no objections if we are moving nemos towards this concept. You might ask why we didn't use it from the beginning on. The reason are again size constraints. You need python in the system for netplan to work.

As we said we don't care for size constraints in the development image I have no problems accepting this change.

However, the more we are working on nemos images the less connection there is with nautilos images. From my
opensource heart I really like what you and Canonical as a company does because it increases the scope of the kiwi builder and proves it can be useful and/or needs adaptions to become useful which is healthy for an opensource project.

From the perspective of the partnership with EB I'm not sure if the nemos image design will help in any case with the nautilos image design because of the differences in almost all areas of the OS. Thus from a strategic point of view we should question ourselves; Does it help EB ? I would say no.

I must admit that I'm not very happy that EB keeps the nautilos distribution derivative closed even to you (our partner). My idea was that we can openly work together which would allow for real collaboration. The situation now is that you are basically forced to build against the open Ubuntu distribution to fulfill the contract with EB. As you cannot look into EB's work, you of course apply all common practices and distribution standards as you do it with all other OS images that you are building. I would have done it in the exact same way.

The problem now is that EB has a different view on many parts that you and me consider common practice. I give you an example. Networking: I suggested let's use systemd(networkd) and udev. EB said: automotive processes requires software verification on all components to pass an audit. systemd is too big and complex to be evaluated, therefore networking has to be provided differently; Solution in nautilos is: netifd. Ok this line of thoughts I could understand but it's not applied to nemos right ? Because in order for you to apply it you have to have access to our repos providing netifd, which you don't have.

This is just one example where the closeness of all this is very painful and I'm afraid it will have some consequences.

Technically I like the solution you provided, so let's go for it

Thanks

**Review by schaefi — APPROVED** [review]

**Comment by IsaacJT** [comment]
Thanks for the feedback and comments. It really is a shame that the end product relies on so many closed-source components. We do have access to the sources on Github for many of them, but of course we can't just package those and include them in NemOS independently. 

> Thus from a strategic point of view we should question ourselves; Does it help EB ? I would say no.

I guess it depends on at what stage you can considering something to stop "helping". Things like this help in that it eases our maintenance burden for generating and testing reference images, which in turn helps us to deliver a more reliable product to you, even if the end reuslt doesn't actually include that utility. 

> As you cannot look into EB's work, you of course apply all common practices and distribution standards as you do it with all other OS images that you are building. I would have done it in the exact same way.

Yes indeed. I think it makes sense for us to do our part in the "Ubuntu way" that we are familiar with and we can reliably test and maintain. 

One thing that occurs to me (which I'm not sure if it's really  helpful or not) is that netplan is made to be as generic as possible which can be used with multiple different network backends (like systemd or NetworkManager). It would definitely be possible to create a netifd backend (or renderer as it's called in netplan-land), allowing us to use one configuration tool with systemd or netifd. How much of a utility this would be I don't know; as you said netplan requires Python, which will likely be removed from any end-images. But it could be an interesting thing to look in to.

**Comment by IsaacJT** [comment]
> You need python in the system for netplan to work.

I asked a colleague about this and Python is only required for the frontend tool (i.e. the `netplan` command). The backend services are all written in C, so netplan can still be used and controlled by restarting services without Python installed.
